.github/workflows/docker.yaml

```diff
@@ -0,0 +1,174 @@
+          push: true
+          no-cache: ${{ steps.apt.outputs.no_cache || github.event_name == 'workflow_dispatch' }}
+          cache-from: type=registry,ref=${{ env.IMAGE }}
+          cache-to: type=inline
+          tags: ${{ env.IMAGE }}
+
+  ci-testing:
+    needs: ci
+    strategy:
+      fail-fast: false
+      matrix:
+        IMAGE: [noetic, master]
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    env:
+      IMAGE: moveit/moveit:${{ matrix.IMAGE }}-${{ github.job }}
+
+    steps:
+      - uses: addnab/docker-run-action@v3
+        name: Check for apt updates
+        id: apt
+        with:
+          image: ${{ env.IMAGE }}
+          run: |
+            apt-get update
+            have_updates=$(apt-get --simulate upgrade | grep -q "^0 upgraded, 0 newly installed, 0 to remove and 0 not upgraded.$" && echo false || echo true)
+            echo "::set-output name=no_cache::$have_updates"
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+        if: ${{ github.event_name == 'workflow_dispatch' || github.event_name != 'schedule' || steps.apt.outputs.no_cache }}
+      - name: Login to Container Registry
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and Push
+        uses: docker/build-push-action@v2
+        if: ${{ github.event_name == 'workflow_dispatch' || github.event_name != 'schedule' || steps.apt.outputs.no_cache }}
+        with:
+          file: .docker/${{ github.job }}/Dockerfile
+          build-args: IMAGE=${{ matrix.IMAGE }}
+          push: true
+          no-cache: ${{ steps.apt.outputs.no_cache || github.event_name == 'workflow_dispatch' }}
+          cache-from: type=registry,ref=${{ env.IMAGE }}
+          cache-to: type=inline
+          tags: |
+            ${{ env.IMAGE }}
+            moveit/moveit:${{ matrix.IMAGE }}-ci-shadow-fixed
+
+  source:
+    needs: ci-testing
+    strategy:
+      fail-fast: false
+      matrix:
+        IMAGE: [noetic, master]
+    runs-on: ubuntu-latest
+    permissions:
+      packages: write
+      contents: read
+    env:
+      IMAGE: moveit/moveit:${{ matrix.IMAGE }}-${{ github.job }}
+
+    steps:
+      - uses: actions/checkout@v2
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+      - name: Login to Container Registry
+        uses: docker/login-action@v1
+        with:
+          username: ${{ secrets.DOCKERHUB_USERNAME }}
+          password: ${{ secrets.DOCKERHUB_TOKEN }}
+      - name: Build and Push
+        uses: docker/build-push-action@v2
+        with:
+          context: .
+          file: .docker/${{ github.job }}/Dockerfile
+          build-args: IMAGE=${{ matrix.IMAGE }}
+          push: true
+          cache-from: type=registry,ref=${{ env.IMAGE }}
+          cache-to: type=inline
+          tags: ${{ env.IMAGE }}
```

Comment:
Remove the docker file since it is not relevant.

Reply:
Done.